pennylane/templates/embeddings.py

```diff
@@ -0,0 +1,250 @@
+
+
+def AmplitudeEmbedding(features, wires=None):
+    """
+    Encodes :math:`2^n` features into the amplitude vector of :math:`n` qubits.
+
+    The absolute square of all  elements in ``features`` has to add up to one.
+
+    .. note::
+
+        AmplitudeEmbedding uses PennyLane's :class:`QubitStateVector` and only works in conjunction with
+        devices that implement this function.
+
+    Args:
+        features (array): Input array of shape ``(2**n, )``
+
+    Keyword Args:
+        wires (Sequence[int]): sequence of qubit indices that the template acts on
+    """
+
+    if not isinstance(wires, Iterable):
+        raise ValueError("Wires needs to be a list of wires that the embedding uses, got {}.".format(wires))
+
+    if 2**len(wires) != len(features):
+        raise ValueError("AmplitudeEmbedding requires a feature vector of size 2**n_qubits which is {}, "
+                         "got {}.".format(2 ** len(wires), len(features)))
+
+    QubitStateVector(features, wires=wires)
+
+
+def BasisEmbedding(features, wires=None):
+    r"""Encodes :math:`n` binary features into a basis state of :math:`n` qubits.
+
+    For example, for ``features=[0, 1, 0]``, the quantum system will be prepared in state :math:`|010 \rangle`.
+
+    .. note::
+
+        BasisEmbedding uses PennyLane's :class:`BasisState` and only works in conjunction with
+        devices that implement this function.
+
+    Args:
+        features (array): Binary input array of shape ``(n, )``
+
+    Keyword Args:
+        wires (Sequence[int]): sequence of qubit indices that the template acts on
+    """
+    if not isinstance(wires, Iterable):
+        raise ValueError("Wires needs to be a list of wires that the embedding uses, got {}.".format(wires))
+
+    if len(features) > len(wires):
+        raise ValueError("Number of bits to embed cannot be larger than number of wires which is {}, "
+                         "got {}.".format(len(wires), len(features)))
+    BasisState(features, wires=wires)
+
+
+def SqueezingEmbedding(features, execution='amplitude', c=0.1, wires=None):
+    r"""Encodes :math:`M` features into the squeezing amplitudes :math:`r` or phases :math:`\phi` of :math:`M` modes.
+
+    The mathematical definition of the squeezing gate is given by the operator
+
+    .. math::
+
+        S(z) = \exp\left(\frac{r}{2}\left(e^{-i\phi}\a^2 -e^{i\phi}{\ad}^{2} \right) \right),
+
+    where :math:`\a` and :math:`\ad` are the bosonic creation and annihilation operators.
+
+    ``features`` has to be an array of ``len(wires)`` floats.
+
+    Args:
+        features (array): Binary sequence to encode
+
+    Keyword Args:
+        execution (str): ``'phase'`` encodes the input into the phase of single-mode squeezing, while
+            ``'amplitude'`` uses the amplitude
+        c (float): parameter setting the value of the phase of all squeezing gates if ``execution='amplitude'``, or the
```

Comment:
parameter setting the value of the phase of all squeezing gates if **execution='phase'**, or the amplitude of all squeezing gates if **execution='amplitude'**

phase/amplitude need to be flipped
also is c=[0,1]? it will be good to mention the range here

Reply:
Strange, in my local version it appears fixed already. 

range for c is now mentioned in the docstring